packages/analytics-js/src/components/core/Analytics.ts

```diff
@@ -445,6 +446,28 @@ class Analytics implements IAnalytics {
       options: payload.options,
       callback: payload.callback,
     });
+
+    // TODO: Maybe we should alter the behavior to send the ad-block page event
+    // even if the SDK is still loaded. It'll be pushed into the to be processed queue.
+
+    // Send automatic ad blocked page event if adblockers are detected on the page
+    if (
+      state.capabilities.isAdBlocked.value === true &&
+      payload.category !== 'RudderJS-Initiated'
```

Comment:
where is this value defined and whats its purpose, is it just to avoid infinite loop of submits? Also can this be done as constant please?

Reply:
Yes, this is to avoid an infinite loop.
I have refactored the code to use constants.